src/core/lib/security/credentials/ssl/ssl_credentials.cc

```diff
@@ -160,6 +200,71 @@ static void ssl_build_server_config(
     config->pem_key_cert_pairs[i].private_key =
         gpr_strdup(pem_key_cert_pairs[i].private_key);
   }
+
+  return config;
+}
+
+void grpc_ssl_server_certificate_config_release(
+    grpc_ssl_server_certificate_config *config) {
+  size_t i;
+
+  if (config == NULL) return;
+
+  for (i = 0; i < config->num_key_cert_pairs; i++) {
+    gpr_free((void *)config->pem_key_cert_pairs[i].private_key);
+    gpr_free((void *)config->pem_key_cert_pairs[i].cert_chain);
+  }
+  gpr_free(config->pem_key_cert_pairs);
+  gpr_free(config->pem_root_certs);
+  gpr_free(config);
+}
+
+grpc_ssl_server_credentials_options *
+grpc_ssl_server_credentials_create_options_using_config(
+    grpc_ssl_client_certificate_request_type client_certificate_request,
+    grpc_ssl_server_certificate_config *config) {
+  grpc_ssl_server_credentials_options *options = NULL;
+
+  if (config == NULL) {
+    gpr_log(GPR_ERROR, "Certificate config must not be NULL.");
+    goto done;
+  }
+
+  options = (grpc_ssl_server_credentials_options *)gpr_zalloc(
+      sizeof(grpc_ssl_server_credentials_options));
+  options->client_certificate_request = client_certificate_request;
+  options->certificate_config = grpc_ssl_server_certificate_config_create(
```

Comment:
`options->certificate_config = config`
No need for a copy here, right?

Reply:
yea makes sense. will just need to remove the `grpc_ssl_server_certificate_config_release(config);` below accordinly